src/background/messages/oauth.ts

```diff
@@ -3,120 +3,43 @@ import browser from "webextension-polyfill";
 import type { PlasmoMessaging } from "@plasmohq/messaging";
 import { Storage } from "@plasmohq/storage";
 
-import type { ColorChatUser, TwitchUser, UserSettings } from "~types/types";
+import type { AccessTokenResponse, TwitchUser, User } from "~types/types";
+import { env } from "~config/env";
+import { getOccupations } from "~services/occupation-service";
 
 const CLIENT_ID = process.env.PLASMO_PUBLIC_TWITCH_CLIENT_ID;
-const TWITCH_API_URL = process.env.PLASMO_PUBLIC_TWITCH_API_URL;
-const API_URL: string = process.env.PLASMO_PUBLIC_API_URL;
 
 const REDIRECT_URI = browser.identity.getRedirectURL();
 
+
 const handler: PlasmoMessaging.MessageHandler = async (req, res) => {
-  const storage = new Storage();
+
   const accessToken = await authenticateWithTwitch();
-  await storage.set("accessToken", accessToken);
+  let [serverAccessToken, user, twitchUser] =
+    await authenticateWithServer(accessToken);
+  user = user as User;
```

Comment:
This is kinda :huh:, but returning as destructured object as suggested may already be enough